kmod/src/inode.c

```diff
@@ -354,16 +355,28 @@ static int set_inode_size(struct inode *inode, struct scoutfs_lock *lock,
 {
 	struct scoutfs_inode_info *si = SCOUTFS_I(inode);
 	struct super_block *sb = inode->i_sb;
+	struct scoutfs_lock *inode_lock = NULL;
+	SCOUTFS_DECLARE_PER_TASK_ENTRY(pt_ent);
 	LIST_HEAD(ind_locks);
 	int ret;
 
 	if (!S_ISREG(inode->i_mode))
 		return 0;
 
-	ret = scoutfs_inode_index_lock_hold(inode, &ind_locks, true, false);
+	ret = scoutfs_lock_inode(sb, SCOUTFS_LOCK_WRITE, 0, inode,
```

Comment:
This is acquiring the lock that is already held and is passed in as an argument.   This might have been the hang that was seen in testing.. blocking on a lock while holding a lock can deadlock with invalidation.   Just use the existing lock arg and add it to the per-task data lock tracker.   I'd add/remove right around the block_truncate_page call to make it clear who's using it.